uaclient/contract.py

```diff
@@ -162,6 +163,29 @@ def request_machine_token_update(
             detach=False,
         )
 
+    def request_machine_activity(
+        self,
+        contract_id: str,
+        machine_id: str,
+        activity_token: str,
+        enabled_services: "List[str]",
+    ):
+        """Report current activity token and enabled services.
+
+        This will report to the contracts backend all the current
+        enabled services in the system.
+        """
+        request_data = {
+            "activity_token": activity_token,
+            "resources": enabled_services,
+        }
+        url = API_V1_MACHINE_ACTIVITY.format(
+            contract=contract_id, machine=machine_id
+        )
+        headers = self.headers()
+
+        return self.request_url(url, headers=headers, data=request_data)
```

Comment:
This response is our machine-token.json so we probably want to persist the token updates inside this method as we do in the UAContractClient attach/update methods instead of outside this method with  self.cfg.write_cache("machine-token", response).

Reply:
I think the only problem of persisting this data here is the problem that we need to update the enabled entitlements for possible resourceToken changes. Do you think I should move that entire logic here ?

Reply:
This might as well be a subset of the machine-token, so we can think about updating only the provided keys.